src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/FormattingTestBase.cs

```diff
@@ -182,6 +182,7 @@ private static (RazorCodeDocument, DocumentSnapshot) CreateCodeDocumentAndSnapsh
             var documentSnapshot = new Mock<DocumentSnapshot>();
             documentSnapshot.Setup(d => d.GetGeneratedOutputAsync()).Returns(Task.FromResult(codeDocument));
             documentSnapshot.Setup(d => d.Project.GetProjectEngine()).Returns(projectEngine);
+            documentSnapshot.Setup(d => d.FilePath).Returns(path);
```

Comment:
This was the biggest bug fix. Without this, we were not generating line pragmas when reparsing.